### DISCUSSION TITLE
About the Implementation Logic of SendToChannel Function

### DISCUSSION THREAD

**Discussion by wyyolo** [FunctionStream/function-stream#148] [Q&A] [answered]
func SendToChannel[T any](ctx context.Context, c chan<- T, e interface{}) bool {
	select {
	case c <- e.(T): // It will panic if `e` is not of type `T` or a type that can be converted to `T`.
		return true
	case <-ctx.Done():
		close(c)
		return false
	}
}

I think we need to implement "case" priority control. If chan has a  buffer, when "c<- e. (T)" and "<- ctx. Done()" both satisfy simultaneously, "select" will randomly execute one of them. What should our goal be?
1. After ctx is cancelled, "c <- e.(T)" can still be executed, and then "<-ctx.Done()" can be executed to close chan. 
2. When ctx is cancelled, priority is given to execution, that is,  "<-ctx.Done()" has a higher priority

**Comment by RobertIndie** [comment, marked as answer]
Just reviewed your code and left some comments: https://github.com/FunctionStream/function-stream/pull/140#pullrequestreview-1892955747

Option 1 sounds good to me. We don't need to give a strong guarantee for the priority. If the `SendToChannel` returns true, that doesn't mean the ctx hasn't done.
